Sources/TuistGenerator/Generator/BuildPhaseGenerator.swift

```diff
@@ -152,7 +152,8 @@ final class BuildPhaseGenerator: BuildPhaseGenerating {
                 shellPath: script.shellPath,
                 shellScript: script.shellScript(sourceRootPath: sourceRootPath),
                 runOnlyForDeploymentPostprocessing: script.runForInstallBuildsOnly,
-                showEnvVarsInLog: script.showEnvVarsInLog
+                showEnvVarsInLog: script.showEnvVarsInLog,
+                dependencyFile: script.dependencyFile?.pathString
```

Comment:
We should use relative here, as we always aim to make the Xcode project portable

Reply:
I am a little confused on how the path resolution works. If I define the path of the dependency file in Project.swift as `$TEMP_DIR/dependencies.d`, I don't want it to be relative to the source directory, since `TEMP_DIR` already contains the entire path.

Reply:
Thats correct and it will work as you suggested, because the "relative" path would be `$TEMP_DIR/dependencies.d`, which is actually not relative.

The problem was that in your tests you were using `/$TEMP_DIR/dependencies.d` (with leading `/`) which as you said is wrong because the variable already contains the entire path

Reply:
If I change `test_generateTarget_action_dependency_file` line 1126 and remove the leading `/` I am getting a precondition failure in Path.swift:700.

Reply:
> Thats correct and it will work as you suggested, because the "relative" path would be $TEMP_DIR/dependencies.d, which is actually not relative.

Doesn't this work on accident though? if `Project.swift` is not in the source directory then `$TEMP_DIR/dependencies.d` will be changed, like it happens in the tests.

In my case when run `generate` the path that I set in Project.swift is changed to `/path/to/project/folder/$TEMP_DIR/dependencies.d` and then changed back to `$TEMP_DIR/dependencies.d` because of the final call to `.relative(to: sourceRootPath).pathString`. Isn't this dance easy to break? Same thing happens for the output files as well.

Reply:
I agree it is, but it's how it works right now and I wouldn't change it as part of this change

Reply:
I agree, I changed the test to use the same root path and not `/test`, now the test passes.